proto/redpanda/core/admin/cluster_link.proto

```diff
@@ -0,0 +1,416 @@
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+
+/* Other Messages */
+/// ClusterLink options
+message ClusterLinkConfiguration {
+    /// Configuration for the internal kafka client
+    ClusterLinkClientOptions client_options = 1;
+    /// Topic metadata sync options
+    /// If not provided, defaults to enabled with no filters
+    optional TopicMetadataSyncOptions topic_metadata_sync_options = 2;
+    /// Consumer offset sync options
+    /// If not provided, defaults to enabled with no filters
+    optional ConsumerOffsetSyncOptions consumer_offset_sync_options = 3;
+    /// Security settings sync options
+    /// If not provided, defaults to enabled with no filters
+    optional SecuritySettingsSyncOptions security_sync_options = 4;
+}
+/// Options for the client link
+message ClusterLinkClientOptions {
+    /// The bootstrap servers to use
+    repeated string bootstrap_servers = 1
+        [(google.api.field_behavior) = REQUIRED];
+    /// The Client ID
+    string client_id = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+    /// TLS settings
+    optional TLSSettings tls_settings = 3;
+    /// Authentication settings
+    optional AuthenticationConfig authentication_config = 4;
+    /// Max metadata age
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 10 seconds
+    int32 metadata_max_age_ms = 5;
+    /// Connection timeout
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 1 second
+    int32 connection_timeout_ms = 6;
+    /// Retry base backoff
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 100ms
+    int32 retry_backoff_ms = 7;
+    /// Fetch request timeout
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 100ms
+    int32 fetch_wait_max_ms = 8;
+    /// Fetch min bytes
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 1 byte
+    int32 fetch_min_bytes = 9;
+    /// Fetch max bytes
+    /// Valid is [1, ...]
+    /// If 0 is provided, defaults to 1MiB
+    int32 fetch_max_bytes = 10;
+}
+/// Options for syncing topic metadata
+message TopicMetadataSyncOptions {
+    /// How often to sync metadata
+    /// If 0 provided, defaults to 30 seconds
+    google.protobuf.Duration interval = 1;
+    /// The topic filters to use
+    repeated NameFilter topic_filters = 2;
+    /// The mirror properties to use.  If not provided will automatically
+    /// mirror:
+    /// - `retention.bytes`
+    /// - `retention.ms`
+    /// - `delete.retention.ms`
+    /// - `min.compaction.lag.ms`
+    /// - `max.compaction.lag.ms`
+    repeated string mirrored_topic_properties = 3;
+    /// Indicates if the mirrored_topic_properties above was provided
+    /// If false, defaults to the above list
+    bool mirrored_topic_properties_provided = 4;
+    bool auto_topic_create_enabled = 5;
```

Comment:
needs docs? I honestly still don't understand this property and don't know why someone would set it false.